### DISCUSSION TITLE
Is it possible to show snaplines for Link also?

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2148] [Q&A]
### Introduction

Is it possible to show snaplines for Link also?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
No, it's currently not possible. What is the use case?

**Reply by reshin-pulapaddy** [reply]
![ezgif com-crop (1)](https://user-images.githubusercontent.com/117301232/234291140-572e59e1-00a0-4e1b-90f8-31c11b55fe05.gif)

when I move the element near to another element in the canvas it is showing the snapline. I need to show the snapline when I move a shape near the link also. is there any way to show Snapline?

**Reply by kumilingus** [reply]
You can show and hide the snaplines programmatically (`show()` and `hide()`). So custom snapping could be an option.
As mentioned [here](https://github.com/clientIO/joint/discussions/2148), it depends on how you use the links. Perhaps they can be just elements that look like a line.

**Reply by reshin-pulapaddy** [reply]
above example, I am using standard.Link

snapline creating code
const snaplines = new joint.ui.Snaplines({
			paper: paper,
		});
		
I tried to show the snapline using snapline.show( ), but it didn't work

**Reply by kumilingus** [reply]
It's currently a protected method. The method usage is this: `this.show({ vertical: 100, horizontal: 100 });`

**Reply by reshin-pulapaddy** [reply]
Hi @kumilingus 
how we can use the protected (show, hide ) method
could you please give an example code

**Reply by kumilingus** [reply]
```ts
// @ts-ignore
snaplines.show({ vertical: 100, horizontal: 100 });
```
Note It only shows the lines, it does not snap the element.

We're now looking into a way for the users to define custom snap points. I'll get back to you.

**Reply by reshin-pulapaddy** [reply]
@kumilingus Thank you

**Reply by kumilingus** [reply]
It will be possible to define additional snap points in version `v3.7` (e.g. you can generate extra points from the links).

**Reply by reshin-pulapaddy** [reply]
Hi @kumilingus 
could you please give an example code

**Reply by kumilingus** [reply]
An example of the `v3.7` functionality? It has not been released yet.
```js
const snaplines = new ui.Snaplines({
  additionalPoints: () => {
     const points = [];
     graph.getLinks().forEach(link => points.push(link.getSourcePoint(), link.getTargetPoint()));
     return points;
  }
})
```

**Reply by reshin-pulapaddy** [reply]
Thanks @kumilingus
